module/applications/advancement/trait-config.mjs

```diff
@@ -103,69 +131,84 @@ export default class TraitConfig extends AdvancementConfig {
     }
     context.default.hint = Trait.localizedList({ grants: this.config.grants, choices: this.config.choices });
 
-    context.choiceOptions = await Trait.choices(this.trait, { chosen, prefixed: true, any: this.selected !== -1 });
-    context.selectedTraitHeader = `${CONFIG.DND5E.traits[this.trait].labels.localization}.other`;
-    context.selectedTrait = this.trait;
+    context.trait = {
+      field: new BooleanField(),
+      input: context.inputs.createCheckboxInput,
+      options: await Trait.choices(this.trait, { chosen, prefixed: true, any: this.selected !== -1 }),
+      selected: this.trait,
+      selectedHeader: `${CONFIG.DND5E.traits[this.trait].labels.localization}.other`,
+      typeField: new StringField({ label: game.i18n.localize("DND5E.ADVANCEMENT.Trait.TraitType") }),
+      typeOptions: Object.entries(CONFIG.DND5E.traits)
+        .filter(([, config]) => ((this.config.mode === "default") || (this.config.mode === "mastery"
+          ? config.mastery : config.expertise)) && (config.dataType !== Number))
+        .map(([value, config]) => ({ value, label: config.labels.title }))
+    };
 
     // Disable selecting categories in mastery mode
     if ( this.advancement.configuration.mode === "mastery" ) {
-      context.choiceOptions.forEach((key, value) => value.disabled = !!value.children);
+      context.trait.options.forEach((key, value) => value.disabled = !!value.children);
     }
 
+    context.mode = {
+      hint: CONFIG.DND5E.traitModes[this.advancement.configuration.mode].hint,
```

Comment:
Throws an error and prevents the sheet from opening again if you set the mode to blank and close the sheet.

Reply:
Blank isn't a valid mode, how are you setting that?

<img width="329" alt="Screenshot 2025-01-17 at 09 55 31" src="https://github.com/user-attachments/assets/9c4a046f-5ba9-4962-811b-6e96393a41c0" />